scheduler/docs/groups.md

```diff
@@ -1,56 +1,106 @@
+parameters, `quantile` and `multiplier` decide when a job is considered a straggler. The
+`quantile` decides the target runtime of the group and the `multipier` decides how many times
+longer than the run time of the quantile job before a job is marked as a straggler.
+
+## Host placement constraints
+
+Sometimes, it is desirable to set some rules on the host distribution to be used for the jobs in a
+group.  For example, jobs in a group might run faster if they are all put on different hosts.
+Another set of jobs may benefit from running on hosts that share the same attribute, such as
+availability zone. Job groups can be configured with host placement constraints to fulfill this
+requirement. Currently, there are three types of host placement constraints:
+
+### Unique host constraint
+Setting this constraint forces the scheduler to place each job in the group on a different machine
+(based on hostname).
+Example:
+```json
+{"groups": [{"uuid": "410198dd-c171-470e-aeb8-6ffa4e6a2ada",
+            "host-placement": {"type": "unique"}],
+ "jobs": [{"uuid": "831836ee-d5c3-47f7-a052-f4c3ce502495",
+           "group": "410198dd-c171-470e-aeb8-6ffa4e6a2ada". ...},
+           ...]}}
+```
+
+
+### Balanced host attribute constraint
+This constraint takes two parameters: an attribute name and a minimum spread. When using it, the
+distribution of hosts will be evenly distributed across values for the named attribute, and the
+minimum size of the distribution will be equal to the minimum spread parameter. For example, given
+"HOSTNAME" as an attribute name and 3 as the minimum spread, jobs will be distributed to a minimum
+of 3 different hosts. If there are 9 jobs in the group, no host may have more than 3 (9 divided by
+3) of the jobs on it. However, the jobs may spread to more than 3 hosts: for example, each job could
+be on a different host.
+Example:
+```json
+{"groups": [{"uuid": "410198dd-c171-470e-aeb8-6ffa4e6a2ada",
+            "host-placement": {"type": "balanced",
+                               "parameters": {"attribute": "HOSTNAME",
+                                              "minimum": 10}}],
+ "jobs": [{"uuid": "831836ee-d5c3-47f7-a052-f4c3ce502495",
+           "group": "410198dd-c171-470e-aeb8-6ffa4e6a2ada". ...},
+           ...]}}
+```
+
+### Attribute equals constraint
+This constraint takes a single parameter: an attribute name. With this constraint, all jobs in a
+group will run in hosts that share the same value for the named attribute. For example, if the
+attribute picked is availability zone, all jobs in the group will be run in hosts that share the
+availability zone attribute attribute value, such as 'us-east-1b'.
+Example:
+```json
+{"groups": [{"uuid": "410198dd-c171-470e-aeb8-6ffa4e6a2ada",
+            "host-placement": {"type": "attribute-equals",
+                               "parameters": {"attribute": "HOSTNAME"}}],
```

Comment:
Why not use availability zone since that's the example in the paragraph above? Also missing one more `}` before the `]`